spec/daru/io/exporters/csv_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'spec_helper'
+require 'tempfile'
+require 'open-uri'
+
+RSpec.describe Daru::IO::Exporters::CSV do
+  context "writes DataFrame to a CSV file" do
+  let(:df) { 
+    Daru::DataFrame.new({
+      'a' => [1,2,3,4,5],
+      'b' => [11,22,33,44,55],
+      'c' => ['a', 'g', 4, 5,'addadf'],
+      'd' => [nil, 23, 4,'a','ff']})
+  }
+  let(:tempfile) { Tempfile.new('data.csv') }
+
+    before { Daru::IO::Exporters::CSV.write df, tempfile.path }
+    subject { Daru::IO::Importers::CSV.load tempfile.path }
```

Comment:
You should DRY it all:
1. Only once `df` defined for all specs (unless somewhere some specific DF is necessary)
2. Only once `tempfile` defined for all specs
3. Only once `before` defined (but passing `params`, by default empty hash, but redefined in necessary contexts)
4. Only once `subject` defined.

This way you can easily focus on specific parts of test inside context.